packages/shared/src/features/onboarding/steps/FunnelContentTypes.tsx

```diff
@@ -0,0 +1,43 @@
+import type { ReactElement } from 'react';
+import React from 'react';
+import type { FunnelStepContentTypes } from '../types/funnel';
+import { FunnelStepTransitionType } from '../types/funnel';
+import { ContentTypes } from '../../../components/onboarding/ContentTypes/ContentTypes';
+import { FunnelStepCtaWrapper } from '../shared/FunnelStepCtaWrapper';
+import { useActions } from '../../../hooks';
+import { ActionType } from '../../../graphql/actions';
+
+function FunnelContentTypes({
+  parameters: { headline, cta },
+  onTransition,
+}: FunnelStepContentTypes): ReactElement | null {
+  const { completeAction } = useActions();
+
+  const handleComplete = () => {
+    completeAction(ActionType.ContentTypes);
+    onTransition({ type: FunnelStepTransitionType.Complete });
+  };
+
+  return (
+    <FunnelStepCtaWrapper
+      cta={{ label: cta || 'Next' }}
+      onClick={handleComplete}
+      containerClassName="flex w-full flex-1 flex-col items-center justify-center overflow-hidden"
+    >
+      <div className="flex w-full flex-col items-center gap-6 p-6 pt-10 tablet:max-w-96">
+        <ContentTypes headline={headline} />
+      </div>
+    </FunnelStepCtaWrapper>
+  );
+}
+
+export default function FunnelContentTypesInner({
+  isActive,
+  ...props
+}: FunnelStepContentTypes): ReactElement | null {
+  if (!isActive) {
+    return null;
+  }
+
+  return <FunnelContentTypes {...props} />;
+}
```

Comment:
Since this is a common use case, maybe we can create some sort of common HOC like this. WDYT?

```tsx
export const withIsActiveGuard = <Step extends FunnelStep>(
  Component: React.FC<Step>,
) => {
  return function GuardedStep(props: Step) {
    const { isActive } = props;
    if (!isActive) {
      return null;
    }
    return <Component {...props} />;
  };
};
```


```suggestion
export const FunnelContentTypes = withIsActiveGuard(FunnelContentTypesInner);
export default FunnelContentTypes;
```

Reply:
Yeah, we can create one like this. I'll add it.